instance/models/openedx_instance.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Open edX Instance models
+"""
+from django.conf import settings
+from django.db import models
+
+from instance.gandi import GandiAPI
+from instance.logging import log_exception
+from .mixins.openedx_database import OpenEdXDatabasesMixin
+from .mixins.version_control import GitHubInstanceMixin
+from .instance import Instance
+from .openedx_appserver import OpenEdXAppConfiguration, OpenEdXAppServer
+
+# Constants ###################################################################
+
+gandi = GandiAPI()
+
+# Models ######################################################################
+
+
+class BaseOpenEdXInstance(Instance, OpenEdXAppConfiguration, OpenEdXDatabasesMixin):
+    """
+    BaseOpenEdXInstance: represents a website or set of affiliated websites powered by the same
+    OpenEdX installation.
+    """
+
+    # Most settings/fields are inherited from OpenEdXAppConfiguration
+    sub_domain = models.CharField(max_length=50)
+    base_domain = models.CharField(max_length=50, blank=True)
+
+    active_appserver = models.OneToOneField(OpenEdXAppServer, null=True, on_delete=models.SET_NULL, related_name='+')
```

Comment:
Why not query this dynamically based on the appserver statuses, rather than a OneToOne field?

Also, we already know that we will need to add support for multi-VM setups. Can we try to make the transition down the line a bit easier - for example, would using an object set rather than a single value be helpful, here?